staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -213,6 +210,10 @@ type GenericAPIServer struct {
 
 	// Version will enable the /version endpoint if non-nil
 	Version *version.Info
+
+	// TerminationSignals provides access to the various termination
+	// signals that happen during the shutdown period of the apiserver.
+	TerminationSignals *TerminationSignals
```

Comment:
is this ever nil?

Reply:
does this have to be public ?

Reply:
it should not be public, i made it private